Specs/Scene/SingleTileImageryProviderSpec.js

```diff
@@ -228,4 +227,45 @@ defineSuite([
             });
         });
     });
+
+    it('options.image can be a HTMLImageElement', function() {
+        var image = new Image(12, 16);
+        var provider = new SingleTileImageryProvider({
+            image : image
+        });
+
+        expect(provider._image).toBe(image);
+        expect(provider._tileWidth).toEqual(image.width);
+        expect(provider._tileHeight).toEqual(image.height);
+        expect(provider._ready).toBe(true);
+    });
+
+    it('options.image can be a HTMLCanvasElement', function() {
+        var canvas = document.createElement('canvas');
+        canvas.width  = 12;
+        canvas.height = 16;
+
+        var provider = new SingleTileImageryProvider({
+            image : canvas
+        });
+
+        expect(provider._image).toBe(canvas);
+        expect(provider._tileWidth).toEqual(canvas.width);
+        expect(provider._tileHeight).toEqual(canvas.height);
+        expect(provider._ready).toBe(true);
+    });
+
+    it('reload calls the _reload function', function() {
```

Comment:
Seems like we would need more unit tests to cover all the cases in the new code, right?